New Cppcheck errors: '[unreadVariable]'

**Checklist**
* [x] I reviewed the [Contributing Guide](https://github.com/nasa/cFS/blob/main/CONTRIBUTING.md).
* [x] I performed a cursory search to see if the bug report is relevant, not redundant, nor in conflict with other tickets.

**Describe the bug**
Latest version of Cppcheck is issuing the following failures for HK:
```
fsw/src/hk_utils.c:57:46: style: Variable 'LastByteAccessed' is assigned a value that is never used. [unreadVariable]
    int32                   LastByteAccessed = 0;
                                             ^
fsw/src/hk_utils.c:126:55: style: Variable 'Loop2' is assigned a value that is never used. [unreadVariable]
    int32                   Loop2                     = 0;
                                                      ^
fsw/src/hk_utils.c:127:55: style: Variable 'MidOfThisPacket' is assigned a value that is never used. [unreadVariable]
    CFE_SB_MsgId_t          MidOfThisPacket           = CFE_SB_INVALID_MSG_ID;
                                                      ^
fsw/src/hk_utils.c:128:55: style: Variable 'SizeOfThisPacket' is assigned a value that is never used. [unreadVariable]
    int32                   SizeOfThisPacket          = 0;
                                                      ^
fsw/src/hk_utils.c:129:55: style: Variable 'FurthestByteFromThisEntry' is assigned a value that is never used. [unreadVariable]
    int32                   FurthestByteFromThisEntry = 0;
                                                      ^
fsw/src/hk_utils.c:130:55: style: Variable 'NewPacketAddr' is assigned a value that is never used. [unreadVariable]
    CFE_SB_Buffer_t *       NewPacketAddr             = 0;
                                                      ^
fsw/src/hk_utils.c:131:55: style: Variable 'Result' is assigned a value that is never used. [unreadVariable]
    int32                   Result                    = CFE_SUCCESS;
                                                      ^
fsw/src/hk_utils.c:273:46: style: Variable 'Loop2' is assigned a value that is never used. [unreadVariable]
    int32                   Loop2            = 0;
                                             ^
fsw/src/hk_utils.c:274:46: style: Variable 'MidOfThisPacket' is assigned a value that is never used. [unreadVariable]
    CFE_SB_MsgId_t          MidOfThisPacket  = CFE_SB_INVALID_MSG_ID;
                                             ^
fsw/src/hk_utils.c:277:46: style: Variable 'Result' is assigned a value that is never used. [unreadVariable]
    int32                   Result           = CFE_SUCCESS;
                                             ^
##[error]Process completed with exit code 255.
```

**To Reproduce**
Run the current version of Cppcheck on the current main branch HK source code.

**Expected behavior**
Cppcheck should pass without raising any errors.

**Reporter Info**
Avi @thnkslprpt

--- a/fsw/src/hk_utils.c
+++ b/fsw/src/hk_utils.c
@@ -54,7 +54,7 @@ void HK_ProcessIncomingHkData(const CFE_SB_Buffer_t *BufPtr)
     uint8 *                 SrcPtr           = NULL;
     size_t                  MessageLength    = 0;
     int32                   MessageErrors    = 0;
-    int32                   LastByteAccessed = 0;
+    int32                   LastByteAccessed;
 
     CFE_MSG_GetMsgId(&BufPtr->Msg, &MessageID);
 
@@ -116,19 +116,19 @@ int32 HK_ValidateHkCopyTable(void *TblPtr)
 /* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 int32 HK_ProcessNewCopyTable(hk_copy_table_entry_t *CpyTblPtr, hk_runtime_tbl_entry_t *RtTblPtr)
 {
-    hk_copy_table_entry_t * StartOfCopyTable          = NULL;
-    hk_copy_table_entry_t * OuterCpyEntry             = NULL;
-    hk_copy_table_entry_t * InnerDefEntry             = NULL;
-    hk_runtime_tbl_entry_t *StartOfRtTable            = NULL;
-    hk_runtime_tbl_entry_t *OuterRtEntry              = NULL;
-    hk_runtime_tbl_entry_t *InnerRtEntry              = NULL;
-    int32                   Loop1                     = 0;
-    int32                   Loop2                     = 0;
-    CFE_SB_MsgId_t          MidOfThisPacket           = CFE_SB_INVALID_MSG_ID;
-    int32                   SizeOfThisPacket          = 0;
-    int32                   FurthestByteFromThisEntry = 0;
-    CFE_SB_Buffer_t *       NewPacketAddr             = 0;
-    int32                   Result                    = CFE_SUCCESS;
+    hk_copy_table_entry_t * StartOfCopyTable = NULL;
+    hk_copy_table_entry_t * OuterCpyEntry    = NULL;
+    hk_copy_table_entry_t * InnerDefEntry    = NULL;
+    hk_runtime_tbl_entry_t *StartOfRtTable   = NULL;
+    hk_runtime_tbl_entry_t *OuterRtEntry     = NULL;
+    hk_runtime_tbl_entry_t *InnerRtEntry     = NULL;
+    int32                   Loop1            = 0;
+    int32                   Loop2;
+    CFE_SB_MsgId_t          MidOfThisPacket;
+    int32                   SizeOfThisPacket;
+    int32                   FurthestByteFromThisEntry;
+    CFE_SB_Buffer_t *       NewPacketAddr;
+    int32                   Result;
 
     /* Ensure that the input arguments are valid */
     if (((void *)CpyTblPtr == NULL) || ((void *)RtTblPtr == NULL))
@@ -270,11 +270,11 @@ int32 HK_TearDownOldCopyTable(hk_copy_table_entry_t *CpyTblPtr, hk_runtime_tbl_e
     hk_runtime_tbl_entry_t *OuterRtEntry     = NULL;
     hk_runtime_tbl_entry_t *InnerRtEntry     = NULL;
     int32                   Loop1            = 0;
-    int32                   Loop2            = 0;
-    CFE_SB_MsgId_t          MidOfThisPacket  = CFE_SB_INVALID_MSG_ID;
-    void *                  OutputPktAddr    = NULL;
-    void *                  SavedPktAddr     = NULL;
-    int32                   Result           = CFE_SUCCESS;
+    int32                   Loop2;
+    CFE_SB_MsgId_t          MidOfThisPacket;
+    void *                  OutputPktAddr = NULL;
+    void *                  SavedPktAddr  = NULL;
+    int32                   Result;
 
     /* Ensure that the input arguments are valid */
     if (((void *)CpyTblPtr == NULL) || ((void *)RtTblPtr == NULL))

